eureka-client/src/main/java/com/netflix/discovery/InstanceInfoReplicator.java

```diff
@@ -0,0 +1,114 @@
+package com.netflix.discovery;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.util.RateLimiter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A task for updating and replicating the local instanceinfo to the remote server. Properties of this task are:
+ * - configured with a single update thread to guarantee sequential update to the remote server
```

Comment:
As this thread executes remote call, it is vulnerable to halts on socket IO operation.